packages/core/lib/commands/test/run.js

```diff
@@ -68,19 +68,50 @@ module.exports = async function (options) {
     inputFile: file
   });
 
-  if (config.networks[config.network]) {
+  /**
+   * If test network exists and provider, host or url doesn't exist, then internal ganache is started with the user specified
+   * configurations in the config.
+   * If development or test network with provider, host or url exists, then it connects to that network specified in the config.
+   * If no network is specified in the config, then internal ganache is started with default configuration.
+   */
+  const configuredNetwork = config.networks[config.network];
+  const testNetworkDefined = configuredNetwork && config.network == "test";
+  const noProviderHostOrUrlConfigured =
+    configuredNetwork &&
+    !configuredNetwork.provider &&
+    !configuredNetwork.host &&
+    !configuredNetwork.url;
+  const port = await require("get-port")();
+  const ipcOptions = { network: "test" };
+  let numberOfFailures;
+
+  if (testNetworkDefined && noProviderHostOrUrlConfigured) {
```

Comment:
This condition fixes the behavior with `test`, but I don't think it preserves the existing behavior in other cases.  To preserve current behavior, it seems to me that this case case should also be invoked if `config.networks[config.netowork]` is undefined.  (This is the reason I hit "request changes".)

Reply:
The last else branch actually checks for the undefined network. @cds-amal and I talked about this and I will make changes in the comments to make it more understandable.

Reply:
Yes, @eggplantzzz . I think you are correct. We can just make it to if/else. I am working on it and testing it. I will make a PR with this change in a few minutes.